packages/lesswrong/components/editor/EditorFormComponent.tsx

```diff
@@ -59,6 +65,88 @@ export const EditorFormComponent = ({form, formType, formProps, document, name,
   const currentEditorType = contents.type || defaultEditorType;
   const showEditorWarning = (formType !== "new") && (initialEditorType !== currentEditorType) && (currentEditorType !== 'ckEditorMarkup')
   
+  // On the EA Forum, our bot checks if posts are potential criticism,
+  // and if so we show a little card with tips on how to make it more likely to go well.
+  const [postFlaggedAsCriticism, setPostFlaggedAsCriticism] = useState<boolean>(false)
+  const [criticismTipsDismissed, setCriticismTipsDismissed] = useState<boolean>(document.criticismTipsDismissed)
+
+  const { mutate: updatePostCriticismTips } = useUpdate({
+    collectionName: "Posts",
+    fragmentName: "PostsEditCriticismTips",
+  })
+  const handleDismissCriticismTips = () => {
+    // hide the card
+    setCriticismTipsDismissed(true)
+    captureEvent('criticismTipsDismissed', {postId: document._id})
+    // make sure not to show the card for this post ever again
+    updateCurrentValues({criticismTipsDismissed: true})
+    if (formType !== 'new' && document._id) {
+      void updatePostCriticismTips({
+        selector: {_id: document._id},
+        data: {
+          criticismTipsDismissed: true
+        }
+      })
+    }
+  }
+  
+  const [checkPostIsCriticism] = useLazyQuery(gql`
+    query getPostIsCriticism($args: JSON) {
+      PostIsCriticism(args: $args)
+    }
+    `, {
+      onCompleted: (data) => {
+        const isCriticism = !!data.PostIsCriticism
+        setPostFlaggedAsCriticism(isCriticism)
+        if (isCriticism && !postFlaggedAsCriticism) {
+          captureEvent('criticismTipsShown', {postId: document._id})
+        }
+      }
+    }
+  )
+  
+  // On the EA Forum, our bot checks if posts are potential criticism,
+  // and if so we show a little card with tips on how to make it more likely to go well.
+  const checkIsCriticism = useCallback((contents: EditorContents) => {
+    // we're currently skipping linkposts, since the linked post's author is
+    // not always the same person posting it on the forum
+    if (
+      !isEAForum ||
+      collectionName !== 'Posts' ||
+      document.isEvent ||
+      document.debate ||
+      document.shortform ||
+      document.url ||
+      criticismTipsDismissed
+    ) return
+
+    checkPostIsCriticism({variables: { args: {
+      title: document.title ?? '',
+      contentType: contents.type,
+      body: contents.value
+    }}})
+  }, [
+    collectionName,
+    document.isEvent,
+    document.debate,
+    document.shortform,
+    document.url,
+    document.title,
+    criticismTipsDismissed,
+    checkPostIsCriticism
+  ])
+
+  // Run this check up to once per 20 min.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const throttledCheckIsCriticism = useCallback(_.throttle(checkIsCriticism, 1000*60*20), [])
```

Comment:
I can see this missing things in the case where e.g.:
 - they open a draft of their post which has some text in it already
 - they paste in the actual body of the post from a google doc

I think using a shorter throttling duration (like 2 minutes) and also checking how much diff there is between now and the last time we checked could be better

Reply:
Good point, thanks! I added in a version with a shorter duration that gets called if there is a significant amount of text added to the post. Could you do a quick review of the [commit](https://github.com/ForumMagnum/ForumMagnum/pull/7332/commits/4514d9beab28921d3edf2a3f75235b0d83e2f156)?

(I just check the lengths instead of trying to do a real diff because I wanted it to be simple/fast, I think this will cover most cases of pasting text in, and I'm ok with missing some cases of ex. replacing a bunch of text.)

Reply:
Seems good 👍